### ISSUE TITLE
IdsDataGrid: Select across pages / contextual toolbar?

### ISSUE THREAD

**Issue by volante007** [infor-design/enterprise-wc#1565]
**Describe the Issue**
Doesn't seem to be a way to allow selecting across pages? I can't find any setting for that, and it doesn't work out of the box (using server side paging). Also, is there any alternative to the contextual toolbar available in 4.x?

Select across pages 4.x
https://design.infor.com/code/ids-enterprise/latest/demo/components/datagrid/example-paging.html
1. Select one row
2. Go to other page
3. Go back to first page
Result: Selection kept
Web component: Selection not kept (or, it seems it's actually selected, but selection column shows unchecked)

Contextual toolbar 4.x
https://design.infor.com/code/ids-enterprise/latest/demo/components/datagrid/example-multiselect.html
1. Select one or more rows
Result: A contextual toolbar appears which contains actions for the selected row(s)

Is there any plan to support these two features, or is there any alternative?

I guess we can achieve/keep the contextual toolbar simply by toggling it ourselves based on the selection, and make it animate as in 4.x.

So, the selection issue is more important out of the two.

**Comment by tmcconechy** [comment]
@volante007 some features we did not add to reduce the amount we support.

1. select across pages -> could probably add, will discuss
2. contextual action toolbar -> was discussed and design doesnt like it, decided to should work with toolbar buttons hiding and showing as per https://github.com/infor-design/enterprise-wc/issues/1029#issuecomment-1680620974 ->. ut we may want an example for this.

**Comment by volante007** [comment]
Hi @tmcconechy , toolbar with actions enabled/disabled based on selections sounds OK to me.

As for select across pages, I found it mentioned in the datagrid TODO (one of the last listed as major):
https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-data-grid/TODO.md

And as I mentioned, it seems to be selected in the API, but not showing as selected in the selection column. So probably a bug rather than feature missing.

Bottom line: I'm hoping it will be fixed as it's listed as a TODO already :)

**Comment by hey-yall** [comment]
Design resources requested via Jira ticket https://jira.infor.com/browse/IDS-1812

**Comment by jbrcna** [comment]
QA Failed. 
verified in https://main.wc.design.infor.com/ids-data-grid/pagination-client-side.html
when using delete/clear button, it only clears/deletes the selected item on a selected page. Also, the text on the top/left of the Datagrid toolbar does not reflect the correct number of records upon deleting/clearing an item.
![Large GIF (1708x398)](https://github.com/infor-design/enterprise-wc/assets/56722184/714e2f86-b5bf-4d3c-ab86-e95ba35820b5)

https://main.wc.design.infor.com/ids-data-grid/tree-grid-virtual-scroll.html
child was not selected.
<img width="1720" alt="image" src="https://github.com/infor-design/enterprise-wc/assets/56722184/04072360-e555-46a0-b6a5-7a3de6ac66cf">

**Comment by tmcconechy** [comment]
@EdwardCoyle want a new issue for that or want to look?

**Comment by tmcconechy** [comment]
@jbrcna lets make a new issue for this point we will have to follow up on it later
